Sources/XCTest/Public/XCTIssue.swift

```diff
@@ -0,0 +1,84 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+/// Encapsulates all data concerning a test failure or other issue.
+public struct XCTIssue: Sendable {
+    /// Types of failures and other issues that can be reported for tests.
+    public enum IssueType: Int, Sendable {
+        /// Issue raised by a failed XCTAssert or related API.
+        case assertionFailure = 0
+
+        /// Issue raised by the test throwing an error in Swift. This could also occur if an Objective C test is implemented in the form `- (BOOL)testFoo:(NSError **)outError` and returns NO with a non-nil out error.
+        case thrownError = 1
+
+        /// Code in the test throws and does not catch an exception, Objective C, C++, or other.
+        case uncaughtException = 2
+
+        /// One of the XCTestCase(measure:) family of APIs detected a performance regression.
+        case performanceRegression = 3
+
+        /// One of the framework APIs failed internally. For example, XCUIApplication was unable to launch or terminate an app or XCUIElementQuery was unable to complete a query.
+        case system = 4
+
+        /// Issue raised when XCTExpectFailure is used but no matching issue is recorded.
+        case unmatchedExpectedFailure = 5
```

Comment:
Should we leave this one out since the `XCTExpectFailure` API isn't implemented?

Reply:
I'm planning to do a basic implementation of that API shortly, so let's leave it in since it will become active at that point.